mixed-content/generic/common.js

```diff
@@ -313,10 +321,18 @@ function createMediaElement(type, media_attrs, source_attrs) {
   var sourceElement = createElement("source", {});
 
   mediaElement.eventPromise = new Promise(function(resolve, reject) {
-    mediaElement.addEventListener("loadeddata", resolve);
-
-    // Notice that the source element will raise the error.
-    sourceElement.addEventListener("error", reject);
+    // Setting a timeout, as some browsers don't fire `error` events.
```

Comment:
This bit is surprising, the error event on source elements is  practically ancient now, I would have expected it to work. Although, I don't see why the error event should fire here, there's a src attribute on the media element so the error event would be on the media element.

Maybe use the getVideoURI() helper that's used in some places?

Reply:
I dug in a bit more; it looks like WebKit isn't firing an `error` event when blocking mixed content. It does fire a `stalled` event at the media element, though, so I think I can hook that rather than adding the timeout for media elements. It's not clear that I can do the same for prefetch, but perhaps I can fail those early with some feature-detection magic. Hrm. Will look into it. :)

> Maybe use the getVideoURI() helper that's used in some places?

These tests are loading a strange python file that streams bits from a real media file, while also poking at wpt's stash API to record whether the file was requested. I don't think it's worth either teaching that helper about the mechanism for these tests, or teaching these tests to look like the file that helper is loading. :)

Reply:
Oh, didn't realize this wasn't just loading a static resource.